### PR TITLE
[V2 API] Implement FetchFromCollection

### PR DESCRIPTION
**Pull request by shifucun**

### CODE REVIEW

File: internal/server/v2/observation/collection.go [hunk 1]
```diff
@@ -32,5 +40,96 @@ func FetchFromCollection(
...
+	btData, err := stat.ReadStatCollection(
+		ctx, store.BtGroup, bigtable.BtObsCollection,
+		ancestor, childType, variables, queryDate,
+	)
+	if err != nil {
+		return nil, err
+	}
+	result := &pbv2.ObservationResponse{
+		ObservationsByVariable: map[string]*pbv2.VariableObservation{},
+		Facets:                 map[string]*pb.StatMetadata{},
```
Comment (Spaceenter) [inline internal/server/v2/observation/collection.go:52]: nit: Maybe in a future PR, we want to rename "StatMetadata" to "Facet", and rename "MetaHash" to "FaceId"?

File: internal/server/v2/observation/collection.go [hunk 2]
```diff
@@ -32,5 +40,96 @@ func FetchFromCollection(
...
+		)
+		if err != nil {
+			return nil, err
+		}
+		for variable, variableData := range moreResult.ObservationsByVariable {
+			result.ObservationsByVariable[variable] = variableData
+		}
+		for facet := range moreResult.Facets {
+			result.Facets[facet] = moreResult.Facets[facet]
+		}
```
Comment (Spaceenter) [inline internal/server/v2/observation/collection.go:132]: nit:

for facet, res := range moreResult.Facets {
			result.Facets[facet] = res
}

File: internal/server/v2/observation/golden/collection_test.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+				continue
+			}
+			if diff := cmp.Diff(resp, &expected, protocmp.Transform()); diff != "" {
+				t.Errorf("payload got diff: %v", diff)
+				continue
+			}
+		}
+	}
+	if err := test.TestDriver(
+		"Series",
```
Comment (Spaceenter) [inline internal/server/v2/observation/golden/collection_test.go:128]: Collection

**Review by Spaceenter — APPROVED** [review]

File: internal/server/v2/observation/collection.go (hunk 1 above)
Comment (shifucun) [reply internal/server/v2/observation/collection.go:52]: very good idea! I will make a follow up PR with that

File: internal/server/v2/observation/collection.go (hunk 2 above)
Comment (shifucun) [reply internal/server/v2/observation/collection.go]: Done

File: internal/server/v2/observation/golden/collection_test.go (hunk 3 above)
Comment (shifucun) [reply internal/server/v2/observation/golden/collection_test.go]: Done

**Pull request merged**